go/apps/api/routes/v2_apis_delete_api/handler.go

```diff
@@ -60,22 +58,19 @@ func (h *Handler) Handle(ctx context.Context, s *zen.Session) error {
 		)
 	}
 
-	err = h.Permissions.Check(
-		ctx,
-		auth.KeyID,
-		rbac.Or(
-			rbac.T(rbac.Tuple{
-				ResourceType: rbac.Api,
-				ResourceID:   "*",
-				Action:       rbac.DeleteAPI,
-			}),
-			rbac.T(rbac.Tuple{
-				ResourceType: rbac.Api,
-				ResourceID:   req.ApiId,
-				Action:       rbac.DeleteAPI,
-			}),
-		),
-	)
+	auth, err = auth.WithPermissions(ctx, rbac.Or(
```

Comment:
what's the point of returning auth here if it's exactly the same type. it just makes the api weird imo.

it's late and I can't put my thoughts into proper words. take this comment more as a todo for me to circle back